### DISCUSSION TITLE
Working around Vite dependency incompatibility with FreeBSD

### DISCUSSION THREAD

**Discussion by namgo** [woodpecker-ci/woodpecker#2846] [Ask a Question] [answered]
Hi all, I'm looking to get the Woodpecker server installed on FreeBSD.

CLI and runner appear to compile without issue, however Vite seems to depend on rollup, which appears to exclusively support linux/windows/mac/android as targets for reasons I don't quite understand.

Vite is a bundler if I understand correctly, is there any way I can work around it?

I'd be happy to make a PR with changes if vague directions can be provided, although I'm not a frontend developer and don't understand modern tooling standards very well.

I don't believe this is a known problem in Woodpecker, but I'm not sure it deserves an Issue as it's a dependency's dependency. https://github.com/woodpecker-ci/woodpecker/discussions/2245 hasn't gained traction, and if supporting non-docker isn't super important I'd understand!

`@rollup/wasm-node` is suggested as an option, and it does provide `bin/rollup` which may work, however I don't understand how to direct vite to use that module instead of the default native rollup - which is definitely my problem, not yours!

`gmake build-server`
```
`(cd web/; pnpm install --frozen-lockfile; pnpm build)`
> vite build --base=/BASE_PATH

/usr/home/nathan/src/woodpecker/web/node_modules/.pnpm/rollup@4.5.0/node_modules/rollup/dist/native.js:31
        throw new Error(
              ^

Error: Your current platform "freebsd" and architecture "x64" combination is not yet supported by the native Rollup build. Please use the WASM build "@rollup/wasm-node" instead.

The following platform-architecture combinations are supported:
android-arm
android-arm64
darwin-arm64
darwin-x64
linux-arm
linux-arm64
linux-arm64 (musl)
linux-x64
linux-x64 (musl)
win32-arm64
win32-ia32
win32-x64
```

As I'm working through this, I've downgraded vite to "4.0.0". It builds! If I find a solution for getting it work on FreeBSD soon I'll follow up more formally!

**Comment by qwerty287** [comment]
If it's working with v4, this looks like bug at vite's end :thinking:

**Reply by namgo** [reply]
I'd be happy to write up a bug report, but I'm not sure what it should say. Do you have any suggestions?

I'll submit something along the lines of "rollup dependency creates platform incompatibilities" but I'm worried it seems opinionated and like a command rather than a bug report.

**Reply by qwerty287** [reply]
> I'll submit something along the lines of "rollup dependency creates platform incompatibilities" but I'm worried it seems opinionated and like a command rather than a bug report.

I don't really think so, of course it's not the best title but I don't really see a different one.

**Comment by anbraten** [comment]
You could bundle on a different arch and just copy the `dist` folder. 

As it seems to be a pretty rare setup, I wouldn't want any special workarounds being added to the core WP repo.

**Reply by namgo** [reply]
> I wouldn't want any special workarounds

Yeah, I agree. The workaround is pretty straightforward (downgrade vite to v4) but I'm not certain it won't impact the UI.

It's definitely something to be aware of for anyone else is looking to build woodpecker server on less mainstream systems.

**Comment by qwerty287** [comment, marked as answer]
Closing as this is old and an upstream issue - probably https://github.com/vitejs/vite/issues/15122
